### ISSUE TITLE
`elan update` fails on Intel Mac during self-update step

### ISSUE THREAD

**Issue by bradheintz** [leanprover/elan#37]
Attempting to run `elan update` fails, and it looks like a configuration issue. In the captured output below, you can see a URL ending in `elan-.tar.gz`, which should probably be `elan-x86_64-apple-darwin.tar.gz`.

Is there a way to update elan in this state, or do I have to re-install?

Details:

elan 1.0.6 (bf510c5db 2021-05-25)

2014 iMac Retina 5K, Intel Core i7
macOS Big Sur 11.2
zsh 5.8

```
» elan show
installed toolchains
--------------------

stable (default)
leanprover-community/lean:3.31.0
leanprover-community/lean:v3.31.0
leanprover/lean4:stable

active toolchain
----------------

stable (default)
Lean (version 3.32.1, commit 35b3a9c4e2d3, Release)

» elan update
info: syncing channel updates for 'stable'
info: latest update on stable, lean version v3.32.1
info: syncing channel updates for 'stable'
info: latest update on stable, lean version v4.0.0-m2
info: checking for self-updates
info: downloading self-update
Error(Download(HttpStatus(404)), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })
error: could not download file from 'https://github.com/leanprover/elan/releases/download/v1.0.7/elan-.tar.gz' to '/var/folders/rz/m8dyhbks07g8_c02jrs9ljhm0000gn/T/elan-update.9AlFuzoiZge2/elan-.tar.gz'
info: caused by: http request returned an unsuccessful status code: 404

» elan -v update
verbose: updating existing install for 'stable'
verbose: toolchain directory: '/Users/bradheintz/.elan/toolchains/stable'
info: syncing channel updates for 'stable'
info: latest update on stable, lean version v3.32.1
verbose: toolchain is already up to date
verbose: updating existing install for 'leanprover/lean4:stable'
verbose: toolchain directory: '/Users/bradheintz/.elan/toolchains/leanprover--lean4---stable'
info: syncing channel updates for 'stable'
info: latest update on stable, lean version v4.0.0-m2
verbose: toolchain is already up to date
info: checking for self-updates
info: downloading self-update
Error(Download(HttpStatus(404)), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })
error: could not download file from 'https://github.com/leanprover/elan/releases/download/v1.0.7/elan-.tar.gz' to '/var/folders/rz/m8dyhbks07g8_c02jrs9ljhm0000gn/T/elan-update.HYQE4XiPKMVn/elan-.tar.gz'
info: caused by: http request returned an unsuccessful status code: 404
```

**Comment by Kha** [comment]
Thanks for the report, should be fixed in 1.0.8 (so you'll need to update by reinstalling)

**Comment by bradheintz** [comment]
Thanks, @Kha.
